app/javascript/packages/transliterable-field-group/index.ts

```diff
@@ -0,0 +1,375 @@
+import { trackError } from '@18f/identity-analytics';
+import { request } from '@18f/identity-request';
+
+/**
+ * Representation of the response from a transliteration API call
+ */
+interface ValidationResponse {
+  success: boolean;
+  data: Record<string, string>;
+}
+
+/**
+ * Internal cache structure for controlling when API call is made
+ * and reusing result
+ */
+interface ValidationResultCache {
+  preValidationValues: Record<string, string>;
+  postValidationValues: Record<string, string>;
+  result: Record<string, string>;
```

Comment:
what's the different btwn `postValidationValues` and `result`?

Reply:
`postValidationValues` here specifically refers to the request payload, and `result` specifically refers to part of the API's response payload.